### PR TITLE
Add set default datasource

### PR DESCRIPTION
**Pull request by zhyuanqi**
### Description

Add the default datasource. So after customer sets the datasource, this will store in the Kibana config
For overview default datasource, it should have feature below:
1. [Done]Customer can set default datasource and it would be saved in the Kibana file under config. 
2. [Done]Once customer sets a datasource as default, they cannot undefault the datasource

### Issues Resolved

https://github.com/opensearch-project/OpenSearch-Dashboards/issues/6058

## Screenshot
https://github.com/opensearch-project/OpenSearch-Dashboards/assets/53279298/14234c4a-e57f-4d72-b381-7cd2295a2ac7

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by bandinib-amzn** [comment]
Looks like there are some lint issues (see Build and test / Build and Verify on Linux CI). Can you fix those please?

File: src/plugins/data_source_management/public/components/edit_data_source/components/header/header.tsx [hunk 1]
```diff
@@ -25,22 +26,51 @@ export const Header = ({
...
     .docTitle.change;
 
   changeTitle(dataSourceName);
 
+  const setDefaultAriaLabel = i18n.translate(
+    'dataSourcesManagement.editDataSource.setDefaultDataSource',
+    {
+      defaultMessage: 'Set as default Data Source.',
+    }
+  );
```
Comment (bandinib-amzn) [inline src/plugins/data_source_management/public/components/edit_data_source/components/header/header.tsx:55]: nit: Set as default data source

File: src/plugins/data_source_management/public/components/edit_data_source/components/header/header.tsx [hunk 2]
```diff
@@ -25,22 +26,51 @@ export const Header = ({
...
+        }}
+        disabled={isDefaultDataSourceState}
+        iconType={isDefaultDataSourceState ? 'starFilled' : 'starEmpty'}
+        aria-label={setDefaultAriaLabel}
+        data-test-subj="datasource-edit-setDefaultDataSource"
+      >
+        {isDefaultDataSourceState ? 'Default' : 'Set as default'}
+      </EuiButtonEmpty>
+    );
+  };
```
Comment (bandinib-amzn) [inline src/plugins/data_source_management/public/components/edit_data_source/components/header/header.tsx:72]: Also should we make UI changes similar to index patterns screen? 
Screenshot 2024-03-18 at 2 39 07 PM

So on index pattern, there is just star and on tool tip we display a message `Set as default index.`. @kgcreative any thoughts?

File: src/plugins/data_source_management/public/components/edit_data_source/components/header/header.tsx [hunk 3]
```diff
@@ -144,6 +174,8 @@ export const Header = ({
       {/* Right side buttons */}
       <EuiFlexItem grow={false}>
         <EuiFlexGroup alignItems="baseline" gutterSize="m" responsive={false}>
+          {/* Test defaultn button */}
```
Comment (bandinib-amzn) [inline src/plugins/data_source_management/public/components/edit_data_source/components/header/header.tsx:177]: nit: spelling mistake

**Comment by bandinib-amzn** [comment]
Is this option only going to be on edit screen? What about while creating data source? Can we set default data source while creating data source? Just to avoid multiple clicks and rendering multiple forms to set default data source.

File: src/plugins/data_source_management/public/components/edit_data_source/components/header/header.tsx (hunk 2 above)
Comment (zhyuanqi) [reply src/plugins/data_source_management/public/components/edit_data_source/components/header/header.tsx:72]: Hi Bandini. THanks for pointing this out. I have checked with Xenia on this. Let me find the Figma link and add it here!

File: src/plugins/data_source_management/public/components/edit_data_source/components/header/header.tsx (hunk 1 above)
Comment (zhyuanqi) [reply src/plugins/data_source_management/public/components/edit_data_source/components/header/header.tsx:55]: Will do

File: src/plugins/data_source_management/public/components/edit_data_source/components/header/header.tsx (hunk 3 above)
Comment (zhyuanqi) [reply src/plugins/data_source_management/public/components/edit_data_source/components/header/header.tsx:177]: Will change it

File: src/plugins/data_source_management/public/components/edit_data_source/edit_data_source.tsx [hunk 4]
```diff
@@ -128,7 +135,9 @@ export const EditDataSource: React.FunctionComponent<RouteComponentProps<{ id: s
           <EditDataSourceForm
```
Comment (BionIT) [inline src/plugins/data_source_management/public/components/edit_data_source/edit_data_source.tsx:135]: can we add tests for this component?

File: CHANGELOG.md [hunk 5]
```diff
@@ -12,6 +12,9 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 
 - Support dynamic CSP rules to mitigate Clickjacking https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5641
 - [CVE-2020-36604] Employ a patched version of hoek `6.1.3` ([#6148](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/6148))
+- [CVE-2023-45857] Bump `axios` from `0.27.2` to `1.6.1` ([#5470](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5470))
```
Comment (ananzh) [inline CHANGELOG.md:15]: changelog updates should not contain other PRs. probably just need a rebase.

File: src/plugins/data_source_management/public/components/edit_data_source/components/edit_form/edit_data_source_form.tsx [hunk 6]
```diff
@@ -400,6 +402,16 @@ export class EditDataSourceForm extends React.Component<EditDataSourceProps, Edi
...
   };
 
+  setDefaultDataSource = async () => {
+    if (this.props.onSetDefaultDataSource) {
+      await this.props.onSetDefaultDataSource();
+    }
+  }
+
+  isDefault =  () => {
+    return this.props.isDefault;
```
[the remark this answers is not in the document]
Comment (xinruiba) [reply src/plugins/data_source_management/public/components/edit_data_source/components/edit_form/edit_data_source_form.tsx:412]: +1

**Comment by seraphjiang** [comment]
> * [P1]The first create datasource should be auto choice as default datasource
> * [P1] If customer deletes a datasource, clients side should check if it is the default datasource, if it is, it would randomly choose a datasource as default datasouce. If no remaining datasource, the defaultDataSource in Config is none.

@zhyuanqi 
this should be out of scope for this PR. could we remove from PR's description. we may track in the meta issue

**Comment by zhyuanqi** [comment]
> Is this option only going to be on edit screen? What about while creating data source? Can we set default data source while creating data source? Just to avoid multiple clicks and rendering multiple forms to set default data source.
Currently, we don't support set default data source while creating data source. I believe it is the same as index pattern as they don't support set as default during creation. Maybe we can add this in the feature

File: src/plugins/data_source_management/public/components/edit_data_source/edit_data_source.tsx (hunk 4 above)
Comment (zhyuanqi) [reply src/plugins/data_source_management/public/components/edit_data_source/edit_data_source.tsx:135]: added!

**Review by BionIT — APPROVED** [review]

**Comment by ZilongX** [comment]
LGTM, just curious - `[Done]Once customer sets a datasource as default, they cannot undefault the datasource` feels like the set default datasource operation being a one shot thing then if customer can not undefault, so would the default update being out of scope if this PR or that's designed on purposes ?

**Comment by ZilongX** [comment]
meanwhile picking up the latest from main and re-running all the checks

**Comment by zhyuanqi** [comment]
> LGTM, just curious - `[Done]Once customer sets a datasource as default, they cannot undefault the datasource` feels like the set default datasource operation being a one shot thing then if customer can not undefault, so would the default update being out of scope of this PR or that's designed on purpose ?

So we always want customer to have a default datasource. Although customer can not undefault datasource, they can default other datasource in order to undefault the previous one. 
We will have two new features about default coming up for next release as
* The first create datasource should be auto choice as default datasource
* If customer deletes a datasource, clients side should check if it is the default datasource, if it is, it would randomly choose a datasource as default datasouce. If no remaining datasource, the defaultDataSource in Config is none.

**Review by xinruiba — APPROVED** [review]

**Review by BionIT — APPROVED** [review]

**Review by Flyingliuhub — APPROVED** [review]

**Pull request merged**